### PR TITLE
[FEATURE] Group/Project Access Requests API

### LINKED ISSUE #315: Add Access Requests API
For my use it would be primarily viewing and approving of group and project access requests.

https://gitlab.com/help/api/access_requests.md

### PR DESCRIPTION
**Pull request by akkee**
Group/Project Access Requests API

Methods to

- List access requests to a group/project

- Make access request to a group/project

- Approve access request to a group/project

- Deny access request to a group/project

Closes #315

### CODE REVIEW

**Comment by akkee** [comment]
Any chance this could be merged @asedge

File: lib/gitlab/client/access_requests.rb [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    #
+    # @example
+    #    Gitlab.approve_project_access_request(1, 1)
+    #    Gitlab.approve_project_access_request(1, 1, 30)
+    #
+    # @param  [Integer, String] :project(required) The ID or name of a project.
+    # @param  [Integer] :user_id(required) The user ID of the access requester
+    # @param  [Integer] :access_level(optional) A valid access level (defaults: 30, developer access level)
+    # @return <Gitlab::ObjectifiedHash] Information about the approved project access request
+    def approve_project_access_request(project, user_id, access_level = '')
```
Comment (asedge) [inline lib/gitlab/client/access_requests.rb]: What if the third argument were an `options` hash?
```
def approve_project_access_request(project, user_id, options = {})
```
Then the assignment on the next line isn't necessary. What do you think?

**Comment by asedge** [comment]
This looks good.  Just a doc change before it gets merged.

**Comment by asedge** [comment]
Thanks!

**Pull request merged**